### ISSUE TITLE
Libgen.io Integration

### ISSUE THREAD

**Issue by almottier** [Readarr/Readarr#848]
**Is your feature request related to a problem? Please describe.**
I would like a libgen.io integration as it is the main source for my ebooks.

**Describe the solution you'd like**
Libgen.io as a new indexer.

**Describe alternatives you've considered**
It looks Jackett is not supporting it. Do you have any recommendation ?

**Additional context**
no

[AB#698](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/698)

**Comment by bakerboy448** [comment]
unlikely

>Adding libgen into automation sounds like an absolute nightmare - there's a lot of good stuff on there, but there is also a massive amount of trash. fixed formats (pdf) auto-converted to reflowable and vice-versa. Fine when uyou snatch manually, as you can check and grab another but when you automate you are liable to end up with a drive full of junk that you never even notice until you come to need it

[Reference](https://discord.com/channels/642870877901422592/642870877901422596/790509495808688128)

**Comment by ddimick** [comment]
Fair point, but would like to mention that Lazylibrarian has this feature and it is working well for me. Pretty much the only reason I still keep LL around, actually.

**Comment by almottier** [comment]
This is the only reason I don't use Readarr...

**Comment by sbernier1** [comment]
I would really like to have this, it's a really huge source of books. I feel like if lazylibrarian uses this, readarr definitely should!

**Comment by bakerboy448** [comment]
A simple +1 on the original post would suffice; comments unrelating to development work or diagnosing are typically unnecessary :)

**Comment by dskvr** [comment]
> Adding libgen into automation sounds like an absolute nightmare - there's a lot of good stuff on there, but there is also a massive amount of trash. fixed formats (pdf) auto-converted to reflowable and vice-versa. Fine when uyou snatch manually, as you can check and grab another but when you automate you are liable to end up with a drive full of junk that you never even notice until you come to need it

Not in my experience, far better than the indexers readarr provides (half of them private!) 

See [libgen.js](https://github.com/dunn/libgen.js/)

Also, it pins on IPFS so finally a ~_darr_~ _arr_ could get an IPFS download client...

**Comment by bakerboy448** [comment]
> Not in my experience, far better than the indexers readarr provides (half of them private!)

because the majority of trackers need Jackett or Prowlarr to be usable in any of the *Arrs

it's a lot of work to maintain the various indexers and doesn't make sense to add them natively to specific apps when other apps exist for those purposes of converting site scrapping or API to a digestible torznab response

**Comment by austinwbest** [comment]
This is better suited as a Prowlarr or Jackett request.

**Comment by ddimick** [comment]
> This is better suited as a Prowlarr or Jackett request.

Libgen offers neither NZBs nor torrents.

Integration of Libgen would get Readarr closer to feature parity with LL.

**Comment by austinwbest** [comment]
I dont think it was every intended to have a feature set compared to anything.

The same can be said for supporting comic formats like Mylar does.

There hasn't been and likely won't be native scraping support added to the arr apps.

Even if someone did it and submitted a prefect working PR, then it is up to the devs to now have to keep it working. Just unlikely to happen.

**Comment by dskvr** [comment]
> Libgen offers neither NZBs nor torrents.

Is that so? :D 

http://libgen.rs/repository_nzb/
http://libgen.rs/repository_torrent/

**Comment by ddimick** [comment]
> > Libgen offers neither NZBs nor torrents.
> 
> Is that so? :D
> 
> http://libgen.rs/repository_nzb/ http://libgen.rs/repository_torrent/

Fair point, now how do I get those to work with Readarr?

**Comment by bakerboy448** [comment]
> how do I get those to work with Readarr?

you don't as they're not supported nor are planned to be supported

**Comment by dskvr** [comment]
> you don't as they're not supported nor are planned to be supported

Readarr: We manage ebook downloads.
Also Readarr: We do not plan on integrating the most thorough collection of ebooks in the world because reasons. 

Airtight logic.

**Comment by travisterrell** [comment]
Just want to echo my support. I'm not 100% sure why there isn't a strong desire to integrate this, with the various excuses. I do, however, completely understand why it hasn't been done yet. With the other similar software projects already existing, it's way easier to include the same sources they did than to incorporate a brand new one.

Still, I want to reiterate my support for this. If I ever get some free-time from work, I'll try to look into what it will take to build an API to interact with Libgen. I get that there's a viewpoint that it should be added into Jackett instead, which is something that would have to be considered. I think it'd have a better chance of actually getting a PRed into Readarr, though, since this would be adding a ton of book-only functionality to Jackett.

(Note on Libgen.is. I was a heavy user of Usenet, Torrents, and IRC for eBooks prior to Libgen, and imho there is just no comparison. Libgen has a super-wide selection, accurate metadata, multiple versions of many books in different formats. (I'm not saying it's perfect, but the former places are honestly kind of a crapshoot, e.g. each torrent creator just uses whatever format they want, possibly with or without accurate metadata** and the nature of torrent search engines basically requires results to be matched using primarily the torrent name. Just not as good of a system, imho (Unless there is some kind of private "RED-quality" torrent provider for books that enforces that stuff. Not one I can find, at the least.)

**(Which...don't get me started on Calibre refusing to provide an option to auto-grab metadata on import, ha.)

**Comment by Visne** [comment]
Implementing this to Jackett/Prowlarr doesn't make a lot of sense to me. These indexers are seperate applications, because you can reuse them multiple times (Sonarr, Radarr, Lidarr etc). Direct downloading however, with some custom API specifically for Libgen, is only useful for Readarr. Therefore it makes no sense to add it to one of these indexers.

There is actually an -arr app with this same situation: [Bazarr](https://github.com/morpheus65535/bazarr). It automatically downloads subtitles for your movies and shows, from websites like [OpenSubtitles](https://www.opensubtitles.org/). They don't use an indexer either, because their specific API implementations only make sense for subtitles.

**Comment by skynet01** [comment]
Any updates on this? 🥰
What's the point of having preset public indexers in Readarr that don't support books really well? This would really increase the app usage without installing any other apps like Jacket.. that some people don't even know exists.

**Comment by thestraycat** [comment]
+1 libgen integration

**Comment by PearsonFlyer** [comment]
Please don't +1 feature requests. You can already thumbs-up them, and it causes unnecessary noise.

The issue is not closed, which means it's under consideration. When someone cares enough to actually sit down to code it, it will get implemented.

**Comment by dmzoneill** [comment]
+1

**Comment by fede843** [comment]
this is a must have, please.

**Comment by jreuben99** [comment]
plus one. must have.

**Comment by LucaSilva-r** [comment]
it would be amazing if it were to be implemented

**Comment by RogueGhost93** [comment]
Libgen and z-lib are where the books are, nothing comes close to them

**Comment by cammclean182** [comment]
+ 1 to add this feature

**Comment by S11-Git** [comment]
+1 its one of the best places to get books

**Comment by bakerboy448** [comment]
Any additional spam comments of +1 or me too may result in this issue being locked - no further warnings will be given

simply +1 the OP.


**No plans to implement this any time soon. No developers exist who wish to take this on. Clearly no one from the community cares enough about this issue to even try to implement it for years. It'll likely be several more years before someone from the community decides to develop this...if ever**
